/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package controller;
import "google/protobuf/wrappers.proto";
import "clusterstate.proto";

option java_multiple_files = true;
option java_package = "org.apache.openwhisk.grpc";
option java_outer_classname = "RoutingProto";

service RoutingService {
  rpc GetInvocationRoute (GetInvocationRouteRequest) returns (GetInvocationRouteResponse) {}

  rpc RoutingUpdateClusterState (UpdateClusterStateRequest) returns (UpdateClusterStateResponse) {}

  rpc GetArrivalInfo (EmptyRequest) returns (GetArrivalInfoResponse) {}
}

message EmptyRequest {} // placeholder type for request

message GetArrivalInfoResponse {
  map<uint32, uint32> query_count_1s = 1;
  map<uint32, uint32> query_count_5s = 2;
}

message GetInvocationRouteRequest {
  string actionName = 1;
}

message GetInvocationRouteResponse {
  int32 invokerInstanceId = 1;
}